File: app/src/main/java/com/example/learn/ui/viewswitcher/ViewSwitcherItemView.java
package com.example.learn.ui.viewswitcher;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.FrameLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.learn.R;

/**
 * @author fqxyi
 * @date 2018/3/23
 */
public class ViewSwitcherItemView extends FrameLayout {

    private TextView title;
    private TextView content;

    public ViewSwitcherItemView(@NonNull Context context) {
        super(context);
        init(context);
    }

    public ViewSwitcherItemView(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    public ViewSwitcherItemView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context);
    }

    private void init(Context context) {
        inflate(context, R.layout.activity_view_switcher_item, this);

        title = (TextView) findViewById(R.id.title);
        content = (TextView) findViewById(R.id.content);

    }

    public void initData(ViewSwitcherBean viewSwitcherBean) {
        if (viewSwitcherBean == null) {
            return;
        }
        title.setText(viewSwitcherBean.title);
        content.setText(viewSwitcherBean.content);
    }

}

File: app/src/main/java/com/example/learn/ui/viewswitcher/ViewSwitcherItemView.kt
package com.example.learn.ui.viewswitcher

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.FrameLayout
import com.example.learn.R
import kotlinx.android.synthetic.main.activity_view_switcher_item.view.*

/**
 * @author fqxyi
 * @date 2018/3/23
 */
class ViewSwitcherItemView @JvmOverloads constructor(
    context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
) : FrameLayout(context, attrs, defStyleAttr) {

    init {
        View.inflate(context, R.layout.activity_view_switcher_item, this)
    }

    fun initData(viewSwitcherBean: ViewSwitcherBean?) {
        if (viewSwitcherBean == null) {
            return
        }
        itemTitle.text = viewSwitcherBean.title
        itemContent.text = viewSwitcherBean.content
    }
}